packages/compass-schema/src/components/schema-toolbar/schema-toolbar.spec.tsx

```diff
@@ -27,28 +32,39 @@ const MockQueryBarPlugin = QueryBarPlugin.withMockServices({
 
 const testErrorMessage =
   'An error occurred during schema analysis: test error msg';
-
-const renderSchemaToolbar = (
-  props: Partial<ComponentProps<typeof SchemaToolbar>> = {}
-) => {
-  const queryBarProps = {};
-  render(
-    <MockQueryBarPlugin {...(queryBarProps as any)}>
-      <SchemaToolbar
-        analysisState="complete"
-        errorMessage={''}
-        isOutdated={false}
-        onAnalyzeSchemaClicked={() => {}}
-        onResetClicked={() => {}}
-        sampleSize={10}
-        schemaResultId="123"
-        {...props}
-      />
-    </MockQueryBarPlugin>
-  );
-};
+const exportSchemaTestId = 'open-schema-export-button';
 
 describe('SchemaToolbar', function () {
+  let defaultPreferences: PreferencesAccess;
+
+  before(async function () {
+    defaultPreferences = await createSandboxFromDefaultPreferences();
+  });
+
+  const renderSchemaToolbar = (
+    props: Partial<ComponentProps<typeof SchemaToolbar>> = {},
+    preferences: PreferencesAccess = defaultPreferences
+  ) => {
+    const queryBarProps = {};
+    render(
+      <PreferencesProvider value={preferences}>
```

Comment:
render method accepts default preferences and returns preferences with the rendering result, you don't need that extra setup here: https://github.com/mongodb-js/compass/blob/main/configs/testing-library-compass/README.md#render--renderwithconnections